caluma/caluma_workflow/domain_logic.py

```diff
@@ -266,3 +266,32 @@ def post_cancel(case, user):
         )
 
         return case
+
+
+class CancelWorkItemLogic:
+    @staticmethod
+    def validate_for_cancel(work_item):
+        if work_item.status != models.WorkItem.STATUS_READY:
+            raise ValidationError("Only READY work items can be cancelled")
+
+    @staticmethod
+    def pre_cancel(validated_data, user):
+        validated_data["status"] = models.WorkItem.STATUS_CANCELED
+        validated_data["closed_at"] = timezone.now()
+        validated_data["closed_by_user"] = user.username
+        validated_data["closed_by_group"] = user.group
+
+        return validated_data
+
+    @staticmethod
+    def post_cancel(work_item, user):
+        # TODO: reopen previous work item(s)?
```

Comment:
We should do this, right? Otherwise the `Case` would be stalled.

Reply:
Or is the idea, that the client reopens the previous WorkItem?

Reply:
Right now we have no way to know what the previous work item is. I think for now the client needs to decide what happens to the workflow..